pkg/commands/up.go

```diff
@@ -0,0 +1,172 @@
+				Env:     velaEnv,
+			}
+
+			return o.Run()
+		},
+	}
+	cmd.SetOut(ioStream.Out)
+	return cmd
+}
+
+type appfileOptions struct {
+	Kubecli client.Client
+	IO      cmdutil.IOStreams
+	Env     *types.EnvMeta
+}
+
+func (o *appfileOptions) Run() error {
+	o.IO.Info("Loading templates ...")
+	tm, err := template.Load()
+	if err != nil {
+		return err
+	}
+
+	o.IO.Info("Parsing vela.yaml ...")
+	appf, err := appfile.Load()
+	if err != nil {
+		return err
+	}
+
+	appConfig := &v1alpha2.ApplicationConfiguration{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1alpha2.ApplicationConfigurationGroupVersionKind.GroupVersion().String(),
+			Kind:       v1alpha2.ApplicationConfigurationKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      appf.Name,
+			Namespace: o.Env.Namespace,
+		},
+	}
+
+	var cfg bytes.Buffer
+	for sname, svc := range appf.GetServices() {
+		build := svc.GetBuild()
+		ctx := &appfile.Context{
+			Data:      map[string]interface{}{"name": sname, "image": build.Image},
+			IO:        o.IO,
+			Namespace: o.Env.Name,
+		}
+
+		o.IO.Infof("\nBuilding service (%s)...\n", sname)
+		if err := build.BuildImage(ctx); err != nil {
+			return err
+		}
+
+		o.IO.Infof("\nRendering component configs for service (%s)...\n", sname)
+		c, err := svc.RenderService(ctx, tm, &cfg)
+		if err != nil {
+			return err
+		}
+		appConfig.Spec.Components = append(appConfig.Spec.Components, *c)
+	}
+
+	b, err := yaml.Marshal(appConfig)
+	if err != nil {
+		return err
+	}
+	cfg.Write(b)
+	cfg.WriteByte('\n')
+
+	deployFilePath := ".vela/deploy.yaml"
+	o.IO.Infof("writing deploy config to (%s)\n", deployFilePath)
+	if err := os.MkdirAll(filepath.Dir(deployFilePath), 0700); err != nil {
+		return err
+	}
+	if err := ioutil.WriteFile(deployFilePath, cfg.Bytes(), 0600); err != nil {
+		return err
+	}
+
+	o.IO.Infof("\nApplying deploy configs ...\n")
+	return o.ApplyAppConfig(appConfig)
+}
+
+// Apply deploy config resources for the app.
+// It differs by create and update:
+// - for create, it displays app status along with information of url, metrics, ssh, logging.
+// - for update, it rolls out a canary deployment and prints its information. User can verify the canary deployment.
+//   This will wait for user approval. If approved, it continues upgrading the whole; otherwise, it would rollback.
+func (o *appfileOptions) ApplyAppConfig(ac *v1alpha2.ApplicationConfiguration) error {
+	key := apitypes.NamespacedName{
+		Namespace: ac.Namespace,
+		Name:      ac.Name,
+	}
+	o.IO.Infof("\nChecking if app has been deployed...\n")
+	var tmpAC v1alpha2.ApplicationConfiguration
+	err := o.Kubecli.Get(context.TODO(), key, &tmpAC)
+	switch {
+	case apierrors.IsNotFound(err):
+		o.IO.Infof("app has not been deployed, creating a new deployment...\n")
+	case err == nil:
+		o.IO.Infof("app existed, updating existing deployment...\n")
+	default:
+		return err
+	}
+	return o.apply(ac)
+}
+
+func (o *appfileOptions) apply(ac *v1alpha2.ApplicationConfiguration) error {
```

Comment:
We currently have a package named application which is the old format of Appfile `https://github.com/oam-dev/kubevela/blob/master/pkg/application/app.go#L33`

We should merge them, because almost every command will result into change of `application` and finally takes effect there.

We should not setup a totally new way just for `vela up`. This will cause lots of confusion.